proposed/logging-generator.md

```diff
@@ -0,0 +1,327 @@
+# Logging Generator for strongly-typed logging messages
+
+## Summary
+
+This is a proposal for a new approach to logging to allow for more easier setup of strongly-typed logging messages. This can be achieved by using a source generator, that gets triggered with a `LoggerMessageAttribute` on partial logging methods, and would be able to either autogenerate the implementation of such method, or produce compile-time diagnostics hinting developers into proper usage of this logging approach.
```

Comment:
Source generator is a solution, and in fact an alternative solution is a language feature.  We should be starting off with a problem statement.  

You mention
> This is a proposal for a new approach to logging to allow for more easier setup of strongly-typed logging messages

This implies the problem is *it's hard to log strongly-typed logging messages*, but it might be bigger than that.  It'd be nice to describe the problem and its characteristics.  Your goals can then talk about how to solve the problem and potentially the prioritization of the goals.

@shirhatti can you share some thoughts on this?

Reply:
Agreed. The samples show the progression but doesn't exactly explain the fact that the source generator is about making structured logging more convenient.